algosdk/future/transaction.py

```diff
@@ -2393,7 +2393,7 @@ def verify(self, message):
                 try:
                     verify_key.verify(message, subsig.signature)
                     verified_count += 1
```

Comment:
@algochoi A couple of questions for my edification:
* For my edification:  How did you become aware of the change?
  * https://github.com/pyca/pynacl/pull/600/ was merged > 1 year ago.  I originally thought perhaps the change is recent.
* What is the policy for locking dependency versions?
  * https://github.com/algorand/py-algorand-sdk/blob/develop/setup.py suggests the version is _not_ locked. 
  *  To avoid future, unvetted API changes, I'm wondering if it's prudent to lock the version.  Though I'm generally unfamiliar with SDK conventions, so I'm raising the question.

Reply:
* Brice noticed it in the repo a while ago and we tracked it down to the `pynacl` change - you can also see some of the discussion here, but some of it was discussed via slack: https://github.com/algorand/py-algorand-sdk/pull/279
* I'm not sure if such convention exists, but locking dependencies sound like a good decision to me. On the other hand, I'm not sure how we will be notified of more critical updates that happen to our dependencies (security patches? I guess if it's a really important patch, the word will get around anyway).

Reply:
Since pip/python does not allow multiple different versions of a package to be installed (unlike JS), the best practice _appears_ to be that runtime dependencies should not be pinned to a specific version, since otherwise it would be nearly impossible to satisfy pinned dependencies when multiple packages are installed. That being said, I think it would be ok to enforce version minimum and/or maximum versions of packages, as long as those are reasonable.

However, for test/development dependencies (the items in our `requirements.txt`), we _should_ pin those, since they are only installed if you're developing on PyTeal, and it's best to have a consistent environment for that.

Reply:
@jasonpaulos Thanks for the context RE: dependency management.  Some of my commentary stems from lack of familiarity with the Python environment.

Based on the above commentary, enforcing a major version lower + upper bound feels prudent.  

I feel more strongly about the upper bound because it's liable to break the build as new software is released.  Though I'll follow the group's convention.

Reply:
I bounded the package versions to some major releases - updated the PR description with links to these as well. It looks like popular packages do this as well:
* [requests](https://github.com/psf/requests/blob/main/setup.py)
* [web3.py](https://github.com/ethereum/web3.py/blob/master/setup.py)

I did enforce a lower bound on `pynacl` version `1.4.0`, which was introduced in 2020, because it introduces the changes that motivated this PR. `pycryptodomex` already had an existing mimum bound on the package of `3.6.0`.

Reply:
@algochoi Looks reasonable to me - thanks for your patience through a couple of iterations.  Resolving.